pkg/cmd/release/create/create.go

```diff
@@ -19,13 +52,320 @@ func NewCmdCreate(f factory.Factory) *cobra.Command {
 	return nil
 }
+
+func askVersion(octopus *octopusApiClient.Client, ask question.Asker, project *projects.Project, channel *channels.Channel) (string, error) {
+	deploymentProcess, err := octopus.DeploymentProcesses.Get(project, "")
+	if err != nil {
+		return "", err
+	}
+
+	template, err := octopus.DeploymentProcesses.GetTemplate(deploymentProcess, channel.ID, "")
+	if err != nil {
+		return "", err
+	}
+
+	var version string
+	if err := ask(&survey.Input{
+		Default: template.NextVersionIncrement,
+		Message: "Version",
+	}, &version); err != nil {
+		return "", err
+	}
+
+	return version, nil
+}
+
+func selectChannel(octopus *octopusApiClient.Client, ask question.Asker, project *projects.Project) (*channels.Channel, error) {
+	existingChannels, err := octopus.Projects.GetChannels(project)
+	if err != nil {
+		return nil, err
+	}
+
+	return question.SelectMap(ask, "Select the channel in which the release will be created", existingChannels, func(p *channels.Channel) string {
+		// TODO is there any possible scenario where p.Channel might not be included in existingChannel?
+		// we should be able to collapse this to a simple "return p.Name"
+		for _, v := range existingChannels {
+			if p.Name == v.Name {
+				return v.Name
+			}
+		}
+
+		return ""
+	})
+}
+
+func findChannel(octopus *octopusApiClient.Client, project *projects.Project, channelName string) (*channels.Channel, error) {
+	foundChannels, err := octopus.Projects.GetChannels(project) // TODO server-side filtering support in future
+	if err != nil {
+		return nil, err
+	}
+	for _, c := range foundChannels { // server doesn't support channel search by exact name so we must emulate it
+		if strings.EqualFold(c.Name, channelName) {
+			return c, nil
+		}
+	}
+	// TODO should we prompt here instead?
+	return nil, fmt.Errorf("no channel found with name of %s", channelName)
+}
+
+func findProject(octopus *octopusApiClient.Client, projectName string) (*projects.Project, error) {
+	// projectsQuery has "Name" but it's just an alias in the server for PartialName; we need to filter client side
+	projectsPage, err := octopus.Projects.Get(projects.ProjectsQuery{PartialName: projectName})
+	if err != nil {
+		return nil, err
+	}
+	for projectsPage != nil && len(projectsPage.Items) > 0 {
+		for _, c := range projectsPage.Items { // server doesn't support channel search by exact name so we must emulate it
+			if strings.EqualFold(c.Name, projectName) {
+				return c, nil
+			}
+		}
+		projectsPage, err = projectsPage.GetNextPage(octopus.Projects.GetClient())
+		if err != nil {
+			return nil, err
+		} // if there are no more pages, then GetNextPage will return nil, which breaks us out of the loop
+	}
+
+	return nil, fmt.Errorf("no project found with name of %s", projectName)
+}
+
+func selectPackageOverrides(octopus *octopusApiClient.Client, ask question.Asker, project *projects.Project, channel *channels.Channel, releaseID string) (string, error) {
```

Comment:
This function is still JB's unmodified prototype code; I haven't cleaned it up, so please disregard any code review concerns you might have here. Will circle back in a future PR when the package override stuff is done properly